### PR TITLE
tidb-binlog: add binlogctl

### PR DESCRIPTION
**Pull request by IANTHEREAL**
this pr is mainly to add binlogctl
* update vendor for binlogctl  (u can ignore it while review)
* remove useless pkg files and codes (keep pkg simple and useful, u can ignore them while review) 
 -- remove pkg/flags (for set env variable)
 -- remove pkg/types (for parser urls, it's too complex)
*  add `binlogctl` and remove `generate_savepoint`

we would mv `binlog-proto` and `binlog-drive` into `tidb-tools/tidb-binlog` directory @july2993

### CODE REVIEW

**Comment by IANTHEREAL** [comment]
@WangXiangUSTC @july2993 PTAL

File: tidb_binlog/binlogctl/config.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	if err != nil {
+		return errors.Trace(err)
+	}
+	// parse command line options
+	if len(cfg.FlagSet.Args()) > 0 {
+		return errors.Errorf("'%s' is not a valid flag", cfg.FlagSet.Arg(0))
+	}
+
+	if cfg.printVersion {
+		fmt.Printf(utils.GetRawInfo("importer"))
```
Comment (WangXiangUSTC) [inline tidb_binlog/binlogctl/config.go]: s/importer/binlogctl/

File: tidb_binlog/binlogctl/nodes.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			default:
+				return errors.NotSupportedf("node %s", kind)
+			}
+		}
+	}
+
+	return errors.NotFoundf("node %s, id %s from etcd %s", kind, nodeID, urls)
+}
+
+// createRegistry returns am ectd registry
```
Comment (WangXiangUSTC) [inline tidb_binlog/binlogctl/nodes.go]: s/am/an/

File: tidb_binlog/node/registry.go [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+	obj := &NodeStatus{
+		NodeID:         nodeID,
+		Host:           host,
+		IsOffline:      true,
+		LatestKafkaPos: latestKafkaPos,
+		LatestFilePos:  latestFilePos,
+		OfflineTS:      latestTS,
+	}
+
+	log.Infof("[pump] %s mark offline information %+v", nodeID, obj)
```
Comment (WangXiangUSTC) [inline tidb_binlog/node/registry.go]: remove `[pump]`

File: tidb_binlog/node/registry.go [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+	if err != nil {
+		return errors.Annotatef(err, "error marshal NodeStatus(%v)", obj)
+	}
+	key := r.prefixed(prefix, nodeID, "object")
+	err = r.client.Create(ctx, key, string(objstr), nil)
+	return errors.Trace(err)
+}
+
+// RefreshNode keeps the heartbeats with etcd
+func (r *EtcdRegistry) RefreshNode(pctx context.Context, prefix, nodeID string, ttl int64, latestFilePos, latestKafkaPos pb.Pos) error {
```
Comment (WangXiangUSTC) [inline tidb_binlog/node/registry.go:176]: this function seems not use

**Review by WangXiangUSTC — COMMENTED** [review]
rest LGTM

File: tidb_binlog/node/registry.go (hunk 4 above)
Comment (IANTHEREAL) [reply tidb_binlog/node/registry.go:176]: Move here to use in other projects

File: tidb_binlog/binlogctl/meta.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+// Meta contains commit TS that can be used to specifies the location of the synchronized data
+// TODO: improve meta later, like adding offset of kafka topic that corresponds to each pump node
+type Meta struct {
+	CommitTS int64 `toml:"commitTS" json:"commitTS"`
+}
+
+// saveMeta saves current tso in meta file.
+func saveMeta(metaFileName string, ts int64, timeZone string) error {
```
Comment (july2993) [inline tidb_binlog/binlogctl/meta.go:90]: give an example about what the file will like, ts is unrelate to timezone,  why need this

Comment (IANTHEREAL) [reply tidb_binlog/binlogctl/meta.go:90]: lol, ts is unrelate to timezone, but various conversions and human-readable are relate to timezone. leave the problem how to use for @WangXiangUSTC (for which consumer).

**Review by july2993 — APPROVED** [review]
LGTM

**Pull request merged**